kibana-reports/public/components/report_definitions/create/create_report_definition.tsx

```diff
@@ -164,33 +229,43 @@ export function CreateReport(props) {
       delete metadata.trigger.trigger_params;
     }
 
-    httpClient
-      .post('../api/reporting/reportDefinition', {
-        body: JSON.stringify(metadata),
-        headers: {
-          'Content-Type': 'application/json',
-        },
-      })
-      .then(async (resp) => {
-        //TODO: consider handle the on demand report generation from server side instead
-        if (metadata.trigger.trigger_type === 'On demand') {
-          let onDemandDownloadMetadata = {
-            query_url: `${
-              metadata.report_params.core_params.base_url
-            }?_g=(time:(from:'${timeRange.timeFrom.toISOString()}',to:'${timeRange.timeTo.toISOString()}'))`,
-            time_from: timeRange.timeFrom.valueOf(),
-            time_to: timeRange.timeTo.valueOf(),
-            report_definition: metadata,
-          };
-          generateReport(onDemandDownloadMetadata, httpClient);
-        }
-        await handleSuccessToast();
-        window.location.assign(`opendistro_kibana_reports#/`);
-      })
-      .catch((error, message) => {
-        console.log('error in creating report definition: ' + error + message);
-        handleErrorToast();
-      });
+    let error = false;
+    await definitionInputValidation(metadata, error).then((response) => {
+      error = response;
+    });
+    if (error) {
+      handleErrorToast();
+      setPreErrorData(metadata);
```

Comment:
Why do we need this?

Reply:
If the user submits an invalid report definition, then the values will all reset in the client-side due to how `createReportDefinitionRequest` is declared and initialized. This saves the state of the data before the submit, and assigns it to `createReportDefinitionRequest` so the values persist

Reply:
So does that mean the input validation only happens when user submit the report, not after he fills in each field?

Reply:
Correct- the `EuiFormRow` has an `isInvalid` flag but it's only set `onSubmit` right now